src/main/java/build/buildfarm/instance/memory/MemoryInstance.java

```diff
@@ -225,7 +225,19 @@ public boolean putOperation(Operation operation) {
       return false;
     }
     String operationName = operation.getName();
-    if (isExecuting(operation)) {
+    if (operation.getDone()) {
+      // destroy requeue timer
+      Watchdog requeuer = requeuers.remove(operationName);
+      if (requeuer != null) {
+        requeuer.stop();
+      }
+      // destroy action timed out failure
+      Watchdog operationTimeoutDelay =
+          operationTimeoutDelays.remove(operationName);
+      if (operationTimeoutDelay != null) {
+        operationTimeoutDelay.stop();
+      }
+    } else if (isExecuting(operation)) {
       requeuers.get(operationName).pet();
```

Comment:
Lol, this is super cute! I love how you renamed the Watchdog refresh() to pet(), makes so much thematic sense!

Reply:
heh, that's how watchdogs were explained to me ages ago in terms of function, you have to pet them to keep them from biting. I felt it made sense both as a dog person and a user of the java functional with a different method for each 'call' (accept, offer, apply...)